### PR TITLE
hoppscotch: 24.3.3-1 -> 24.8.1-0

### PR DESCRIPTION
**Pull request by DataHearth**
## Description of changes

Update <https://github.com/hoppscotch/hoppscotch> to latest version

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by DataHearth** [comment]
Result of `nixpkgs-review pr 340777` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hoppscotch</li>
  </ul>
</details>

File: pkgs/by-name/ho/hoppscotch/package.nix [hunk 1]
```diff
@@ -1,30 +1,35 @@
...
-, fetchurl
-, appimageTools
-, undmg
+{
+  lib,
+  stdenv,
+  fetchurl,
+  appimageTools,
+  undmg,
+  copyDesktopItems,
```
Comment (GetPsyched) [inline pkgs/by-name/ho/hoppscotch/package.nix:7]: Unused import.
```suggestion
```

File: pkgs/by-name/ho/hoppscotch/package.nix [hunk 2]
```diff
@@ -34,41 +39,56 @@ let
     '';
     homepage = "https://hoppscotch.com";
     downloadPage = "https://hoppscotch.com/downloads";
-    changelog = "https://hoppscotch.com/changelog";
-    license = licenses.mit;
-    maintainers = with maintainers; [ DataHearth ];
+    changelog = "https://github.com/hoppscotch/hoppscotch/releases/tag/2024.8.1";
```
Comment (GetPsyched) [inline pkgs/by-name/ho/hoppscotch/package.nix:42]: The version here should have been dynamic with some string interpolation on the `version` binding.

**Review by GetPsyched — COMMENTED** [review]
This wasn't just a version bump, there are some refactor bits in here; shouldn't have done this in a single commit.

File: pkgs/by-name/ho/hoppscotch/package.nix (hunk 1 above)
Comment (pbsds) [reply pkgs/by-name/ho/hoppscotch/package.nix:7]: https://github.com/NixOS/nixpkgs/pull/343315, pre-existing issue, but i fixed it too

File: pkgs/by-name/ho/hoppscotch/package.nix (hunk 2 above)
Comment (pbsds) [reply pkgs/by-name/ho/hoppscotch/package.nix:42]: https://github.com/NixOS/nixpkgs/pull/343315

**Pull request merged**